packages/aws-cdk-lib/pipelines/lib/codepipeline/codepipeline.ts

```diff
@@ -388,7 +387,7 @@ export class CodePipeline extends PipelineBase {
     this.selfMutationEnabled = props.selfMutation ?? true;
     this.dockerCredentials = props.dockerCredentials ?? [];
     this.singlePublisherPerAssetType = !(props.publishAssetsInParallel ?? true);
-    this.cliVersion = props.cliVersion ?? preferredCliVersion();
+    this.cliVersion = props.cliVersion ?? 'latest';
```

Comment:
If you are sure it is safe to unlock all CLI versions (both `aws-cdk` and `cdk-assets`), then remove the `this.cliVersion` property entirely.

Otherwise, keep the versioning on `aws-cdk` and remove its use only for `cdk-assets`.